frontend/middleware.ts

```diff
@@ -0,0 +1,11 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+import { PROD_HOSTNAME } from '@config/constants';
+
+export function middleware(request: NextRequest) {
+   const headers =
+      request.nextUrl.hostname === PROD_HOSTNAME
+         ? { 'X-Robots-Tag': 'noindex, nofollow' }
+         : undefined;
+   return NextResponse.next({ headers });
```

Comment:
I think this is the format it wants
```suggestion
   const init =
      request.nextUrl.hostname === PROD_HOSTNAME
         ? { headers: ['X-Robots-Tag', 'noindex, nofollow'] }
         : undefined;
   return NextResponse.next(init);
```

Reply:
I think they're equivalent based on the definition of HeadersInit.

`type HeadersInit = [string, string][] | Record<string, string> | Headers;`

But I'm happy to use the array syntax as well.